.github/workflows/ci.yml

```diff
@@ -0,0 +1,86 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+  schedule:
+    # Run at 18:00 UTC every day
+    - cron: '0 18 * * *'
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # Set 0 to rebase later.
+          fetch-depth: 0
+
+      - name: Read .env
+        run: |
+          . ./.env
+          echo "MDBOOK_VERSION=${MDBOOK_VERSION}" >> $GITHUB_ENV
+          echo "MDBOOK_LINKCHECK_VERSION=${MDBOOK_LINKCHECK_VERSION}" >> $GITHUB_ENV
+          echo "MDBOOK_TOC_VERSION=${MDBOOK_TOC_VERSION}" >> $GITHUB_ENV
+          echo "DEPLOY_DIR=${DEPLOY_DIR}" >> $GITHUB_ENV
+
+      - name: Cache binaries
+        id: mdbook-cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin
+          key: ${{ runner.os }}-${{ env.MDBOOK_VERSION }}--${{ env.MDBOOK_LINKCHECK_VERSION }}--${{ env.MDBOOK_TOC_VERSION }}
+
+      - name: Cache linkcheck
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/book/linkcheck
+          key: ${{ runner.os }}-${{ hashFiles('./book/linkcheck') }}
+
+      - name: Check line lengths
+        if: github.event_name != 'schedule'
+        run: |
+          shopt -s globstar
+          MAX_LINE_LENGTH=100 bash ci/check_line_lengths.sh src/**/*.md
+
+      - name: Install latest nightly Rust toolchain
+        if: steps.mdbook-cache.outputs.cache-hit != 'true'
+        uses: actions-rs/toolchain@v1
+        with:
+            toolchain: nightly
+            override: true
+
+      - name: Install Dependencies
+        if: steps.mdbook-cache.outputs.cache-hit != 'true'
+        run: |
+          cargo install mdbook --version ${{ env.MDBOOK_VERSION }}
+          cargo install mdbook-linkcheck --version ${{ env.MDBOOK_LINKCHECK_VERSION }}
+          cargo install mdbook-toc --version ${{ env.MDBOOK_TOC_VERSION }}
```

Comment:
I haven't come up with a good way to use `--version ^x.y.z` while using actions/cache.

Reply:
This seems fine to me.

Reply:
I guess that means we'll have to manually update these tools more often?

Reply:
Is it worth it to introduce some automation to update those dependencies?

Reply:
Well, I guess we could might be able to get dependabot setup, but I'm not sure if it's worth doing it here.

Reply:
AFAIK dependabot only updates Cargo.toml or Cargo.lock so it doesn't help us. But an automation job or something else would be great as follow-up work.

> I guess that means we'll have to manually update these tools more often?

Yes, we _can_ still use `--version ^x.y.z` specifying but it doesn't update the mdbook and their tools version due to the cache key.